react/fields/Dropdown/Dropdown.js

```diff
@@ -77,10 +77,6 @@ export default class Dropdown extends Component {
     options: PropTypes.arrayOf((propValue, key, componentName) => {
       const { value, label } = propValue[key];
 
-      if (typeof value !== 'string') {
-        return new Error(`Invalid prop \`options[${key}].value\` of type \`${typeof value}\` supplied to \`${componentName}\`, expected \`string\`.`);
-      }
-
```

Comment:
Rather than removing this check we should redefine our expectations here to something like:
```js
options: PropTypes.arrayOf(
  PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.object),
      PropTypes.string
    ]).isRequired,
    label: PropTypes.string
  })
)
```